### DISCUSSION TITLE
Unable to use .war from this fork

### DISCUSSION THREAD

**Discussion by gorman42** [kagemomiji/airsonic-advanced#191] [Q&A] [answered]
Hi! I'm currently running 11.0.0-SNAPSHOT.20230217142243 – February 17, 2023 at 3:22:43 PM CET [Commit: [1397446f979b1cdea283eec89ce4f0eae7d63450](https://github.com/airsonic-advanced/airsonic-advanced/commit/1397446f979b1cdea283eec89ce4f0eae7d63450)]

I tried downloading the stable release from your repository, substituting the .war file but I got these errors:

`sudo systemctl status airsonic.service
× airsonic.service - Airsonic Media Server
     Loaded: loaded (/etc/systemd/system/airsonic.service; enabled; vendor preset: enabled)
     Active: failed (Result: exit-code) since Thu 2023-04-20 04:06:35 CEST; 4s ago
    Process: 196699 ExecStart=/usr/bin/java $JAVA_OPTS -Dairsonic.home=${AIRSONIC_HOME} -Dserver.servlet.context-path=${CONTEXT_PATH} ->
   Main PID: 196699 (code=exited, status=1/FAILURE)
        CPU: 483ms

Apr 20 04:06:35 gorman-NUC java[196699]:         at org.springframework.boot.loader.LaunchedURLClassLoader.loadClass(LaunchedURLClassLo>
Apr 20 04:06:35 gorman-NUC java[196699]:         at java.base/java.lang.ClassLoader.loadClass(ClassLoader.java:522)
Apr 20 04:06:35 gorman-NUC java[196699]:         at java.base/java.lang.Class.forName0(Native Method)
Apr 20 04:06:35 gorman-NUC java[196699]:         at java.base/java.lang.Class.forName(Class.java:398)
Apr 20 04:06:35 gorman-NUC java[196699]:         at org.springframework.boot.loader.MainMethodRunner.run(MainMethodRunner.java:46)
Apr 20 04:06:35 gorman-NUC java[196699]:         at org.springframework.boot.loader.Launcher.launch(Launcher.java:108)
Apr 20 04:06:35 gorman-NUC java[196699]:         at org.springframework.boot.loader.Launcher.launch(Launcher.java:58)
Apr 20 04:06:35 gorman-NUC java[196699]:         at org.springframework.boot.loader.WarLauncher.main(WarLauncher.java:59)
Apr 20 04:06:35 gorman-NUC systemd[1]: airsonic.service: Main process exited, code=exited, status=1/FAILURE
Apr 20 04:06:35 gorman-NUC systemd[1]: airsonic.service: Failed with result 'exit-code'.
`

**Comment by kagemomiji** [comment]
@gorman42 
Could you show me your `airsonic.service`, the output of `java --version` and your OS information?
I suppose there is `airsonic.log` under `/var/airsonic` or `C:\airsonic`. Could you upload it?

**Comment by gorman42** [comment]
Absolutely.

Airsonic service:

```
[Unit]
Description=Airsonic Media Server
After=remote-fs.target network.target
AssertPathExists=/var/airsonic

[Service]
Type=simple
Environment="JAVA_JAR=/var/airsonic/airsonic.war"
Environment="JAVA_OPTS=-Xmx700m"
Environment="AIRSONIC_HOME=/var/airsonic"
Environment="PORT=41348"
# Environment="CONTEXT_PATH=/airsonic"
Environment="JAVA_ARGS="
EnvironmentFile=-/etc/sysconfig/airsonic
ExecStart=/usr/bin/java \
          $JAVA_OPTS \
          -Dairsonic.home=${AIRSONIC_HOME} \
          -Dserver.servlet.context-path=${CONTEXT_PATH} \
          -Dserver.port=${PORT} \
          -jar ${JAVA_JAR} $JAVA_ARGS
User=airsonic
Group=airsonic

# See https://www.freedesktop.org/software/systemd/man/systemd.exec.html
# for details
DevicePolicy=closed
NoNewPrivileges=yes
PrivateDevices=yes
PrivateTmp=yes
PrivateUsers=yes
ProtectControlGroups=yes
ProtectKernelModules=yes
ProtectKernelTunables=yes
RestrictAddressFamilies=AF_UNIX AF_INET AF_INET6
RestrictNamespaces=yes
RestrictRealtime=yes
SystemCallFilter=~@clock @debug @module @mount @obsolete @privileged @reboot @setuid @swap
ReadWritePaths=/var/airsonic

# You can change the following line to `strict` instead of `full`
# if you don't want airsonic to be able to
# write anything on your filesystem outside of AIRSONIC_HOME.
ProtectSystem=full

# You can uncomment the following line if you don't have any media
# in /home/…. This will prevent airsonic from ever reading/writing anything there.
#ProtectHome=true

# You can uncomment the following line if you're not using the OpenJDK.
# This will prevent processes from having a memory zone that is both writeable
# and executeable, making hacker's lifes a bit harder.
#MemoryDenyWriteExecute=yes


[Install]
WantedBy=multi-user.target
```

----------

Java version:

openjdk 11.0.18 2023-01-17
OpenJDK Runtime Environment (build 11.0.18+10-post-Ubuntu-0ubuntu122.04)
OpenJDK 64-Bit Server VM (build 11.0.18+10-post-Ubuntu-0ubuntu122.04, mixed mode)

I attach the log. I tried renaming it to .bak when starting anew with your war file, but it did not even arrive to the point where a log was created.


[airsonic.log](https://github.com/kagemomiji/airsonic-advanced/files/11285797/airsonic.log)

**Reply by kagemomiji** [reply, marked as answer]
@gorman42 
Could you try java17? 
The war file of this repository was compiled with Java17, so it is not guaranteed to run on Java 11.

**Comment by gorman42** [comment]
Updated Java. Below the results.

```
java -version
openjdk version "17.0.6" 2023-01-17
OpenJDK Runtime Environment (build 17.0.6+10-Ubuntu-0ubuntu122.04)
OpenJDK 64-Bit Server VM (build 17.0.6+10-Ubuntu-0ubuntu122.04, mixed mode, sharing)
```

```
sudo systemctl status airsonic.service

× airsonic.service - Airsonic Media Server
     Loaded: loaded (/etc/systemd/system/airsonic.service; enabled; vendor preset: enabled)
     Active: failed (Result: signal) since Thu 2023-04-20 14:34:32 CEST; 58s ago
    Process: 322450 ExecStart=/usr/bin/java $JAVA_OPTS -Dairsonic.home=${AIRSONIC_HOME} -Dserver.servlet.context-path=${CONTEXT_PATH} -Dserver.port=${PORT} -jar ${JAVA_JAR} $JA>
   Main PID: 322450 (code=killed, signal=SYS)
        CPU: 5.187s

Apr 20 14:34:30 gorman-NUC java[322450]:     /  \   _ _ __  ___  ___  _ __  _  ___
Apr 20 14:34:30 gorman-NUC java[322450]:    / /\ \ | | '__|/ __|/ _ \| '_ \| |/ __|
Apr 20 14:34:30 gorman-NUC java[322450]:   / ____ \| | |   \__ \ (_) | | | | | (__
Apr 20 14:34:30 gorman-NUC java[322450]:  /_/    \_\_|_|   |___/\___/|_| |_|_|\___|
Apr 20 14:34:30 gorman-NUC java[322450]:                                                
Apr 20 14:34:30 gorman-NUC java[322450]:                         11.1.1
Apr 20 14:34:30 gorman-NUC java[322450]: 2023-04-20 14:34:30.262  WARN --- o.a.p.u.LegacyHsqlMigrationUtil          : HSQLDB database upgrade needed, from version 2.5.0 to 2.7
Apr 20 14:34:32 gorman-NUC systemd[1]: airsonic.service: Main process exited, code=killed, status=31/SYS
Apr 20 14:34:32 gorman-NUC systemd[1]: airsonic.service: Failed with result 'signal'.
Apr 20 14:34:32 gorman-NUC systemd[1]: airsonic.service: Consumed 5.187s CPU time.


```

**Comment by gorman42** [comment]
Managed to solve it through suggestion here: https://github.com/airsonic-advanced/airsonic-advanced/issues/617#issuecomment-1095293240

**Comment by gorman42** [comment]
I hope that, by keeping to stable releases of yours, from now on, I won't meet other problems. Thanks for continuing development on this. It's much appreciated. :-)
